Shell/Builtin.cpp

```diff
@@ -0,0 +1,743 @@
+/*
+ * Copyright (c) 2020, The SerenityOS developers.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "Shell.h"
+#include <AK/LexicalPath.h>
+#include <LibCore/ArgsParser.h>
+#include <signal.h>
+#include <sys/wait.h>
+
+extern RefPtr<Line::Editor> editor;
+
+int Shell::builtin_alias(int argc, const char** argv)
+{
+    Vector<const char*> arguments;
+
+    Core::ArgsParser parser;
+    parser.add_positional_argument(arguments, "List of name[=values]'s", "name[=value]", Core::ArgsParser::Required::No);
+
+    if (!parser.parse(argc, const_cast<char**>(argv), false))
+        return 1;
+
+    if (arguments.is_empty()) {
+        for (auto& alias : m_aliases)
+            printf("%s=%s\n", alias.key.characters(), alias.value.characters());
+        return 0;
+    }
+
+    bool fail = false;
+    for (auto& argument : arguments) {
+        auto parts = String { argument }.split_limit('=', 2, true);
+        if (parts.size() == 1) {
+            auto alias = m_aliases.get(parts[0]);
+            if (alias.has_value()) {
+                printf("%s=%s\n", parts[0].characters(), alias.value().characters());
+            } else {
+                fail = true;
+            }
+        } else {
+            m_aliases.set(parts[0], parts[1]);
+        }
+    }
+
+    return fail ? 1 : 0;
+}
+
+int Shell::builtin_bg(int argc, const char** argv)
+{
+    int job_id = -1;
+
+    Core::ArgsParser parser;
+    parser.add_positional_argument(job_id, "Job id to run in background", "job_id", Core::ArgsParser::Required::No);
```

Comment:
```suggestion
    parser.add_positional_argument(job_id, "Job ID to run in background", "job-id", Core::ArgsParser::Required::No);
```

(In command line, `we-use-this-case` `instead_of_this_case`)